src/index.js

```diff
@@ -51,7 +51,7 @@ const Alpine = {
 
         const observerOptions = {
             childList: true,
-            attributes: true,
+            attributeFilter: ['x-data'],
```

Comment:
We need to initialize only elements with `x-data` attribute on them to initialize the component.

By setting the option `attributeFilter` we're incredibly reducing complexity of the MutationObserver by triggering the callback only when we confident that element has `x-data` attribute and then we run nested loops.

We should avoid executing nested loops as much as we can especially when we they are useless.